### PR TITLE
Add ESTree AST and Acorn definitions.

### PR DESCRIPTION
**Pull request by RReverser**
- Adds type defs for [ESTree AST specification](https://github.com/estree/estree) for ES6.
- Updates Esprima to use it since it follows this spec.
- Adds Acorn type definitions (it follows the same spec).

Also likely supersedes #3201 with non-Mozilla-specific and up-to-date definitions, but didn't remove it for now.

### CODE REVIEW

**Comment by vvakame** [comment]
@teppeis could you review this PR?

File: estree/estree.d.ts [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+  }
+
+  interface ConditionalExpression extends Expression {
+    test: Expression;
+    alternate: Expression;
+    consequent: Expression;
+  }
+
+  interface CallExpression extends Expression {
+    callee: Expression;
```
Comment (teppeis) [inline estree/estree.d.ts]: @RReverser `callee: Expression | Super` is correct?
https://github.com/estree/estree/blob/master/es6.md#expressions

Comment (RReverser) [inline estree/estree.d.ts]: @teppeis Yes, why?

Comment (RReverser) [inline estree/estree.d.ts]: `callee: Super` is for cases like in
```
class A extends B {
  constructor() {
    super(); // <-- this is CallExpression(callee: Super)
    ...
  }
}
```

Comment (teppeis) [inline estree/estree.d.ts]: @RReverser Yes, but the line:197 is `callee: Expression` not `callee: Expression | Super`.
Are you fixing it?

Comment (RReverser) [inline estree/estree.d.ts]: Ah, I see it, thanks. Probably something in generator is broken, let me look into it.

Comment (RReverser) [inline estree/estree.d.ts]: Yeah, parser didn't recognize multiple extensions of same interface in same spec (this was a case only for `CallExpression` and only in ES6). Thanks, fixed now.

**Comment by teppeis** [comment]
@RReverser @vvakame LGTM! thanks!

**Comment by RReverser** [comment]
@tappeis @vvakame Thanks for the review! Any thoughts on whether it's worth to supersede #3201 by this one?

**Comment by teppeis** [comment]
@RReverser I think #3201 is deprecated by this estree.d.ts and it's ok to remove if @vvakame agrees.

**Comment by vvakame** [comment]
@teppeis thanks!
@RReverser thanks mate!

**Comment by RReverser** [comment]
Cool, thanks guys!

**Pull request merged**